LibGit2Sharp/Branch.cs

```diff
@@ -226,7 +236,17 @@ private Branch ResolveTrackedBranch()
                 return null;
             }
 
-            string trackedReferenceName = Proxy.git_branch_upstream_name(repo.Handle, CanonicalName);
+            string trackedReferenceName = null;
+            try
+            {
+                trackedReferenceName = Proxy.git_branch_upstream_name(repo.Handle, CanonicalName);
+            }
+            catch (InvalidSpecificationException)
+            {
+                // we have no way to pro-actively check this
```

Comment:
How about something like `git_branch_upstream_name throws InvalidSpecificationException when a user sets the remote to be a full URL`. But won't this mean that we return `null` even though we're tracking a branch?